### PR TITLE
TAO-10423 added EventAggregatorService

### PR DESCRIPTION
**Pull request by andreluizmachado**
Triggering the aggregated events after the action is dispatched.

Please check the PR https://github.com/oat-sa/generis/pull/826 for more details.

### CODE REVIEW

File: migrations/Version202007231447592234_tao.php [hunk 1]
```diff
@@ -31,6 +35,16 @@ public function up(Schema $schema): void
         $this->getServiceManager()->register(EventManager::SERVICE_ID, $eventManager);
         $this->getServiceManager()->register(ClassPropertiesChangedListener::SERVICE_ID, new ClassPropertiesChangedListener());
         $this->getServiceManager()->register(ClassPropertyRemovedListener::SERVICE_ID, new ClassPropertyRemovedListener());
+
+        $this->getServiceManager()->register(IndexUpdaterInterface::SERVICE_ID, new GenerisIndexUpdater());
```
Comment (luisfn) [inline migrations/Version202007231447592234_tao.php:39]: Don't we need to register these for install as well?

File: models/classes/search/strategy/GenerisIndexUpdater.php [hunk 2]
```diff
@@ -38,6 +38,6 @@ public function deleteProperty(array $property): void
 
     public function hasClassSupport(string $class): bool
     {
-        return true;
+        return false;
```
Comment (Skyld) [inline models/classes/search/strategy/GenerisIndexUpdater.php:41]: Shouldn't we add a test case to verify this behavior?

**Review by Skyld — CHANGES_REQUESTED** [review]

File: models/classes/search/strategy/GenerisIndexUpdater.php (hunk 2 above)
Comment (andreluizmachado) [reply models/classes/search/strategy/GenerisIndexUpdater.php:41]: Hi, unit tests were added.

**Review by Skyld — APPROVED** [review]
- [x] New code is covered by tests
- [ ] Tests are running successfully (old and new ones) on my local machine
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues
- [ ] Feature is working correctly on my local machine
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes
- [ ] Documentation is updated according to changes

**Pull request merged**